source/h1_connection.c

```diff
@@ -1098,7 +1096,17 @@ static int s_decoder_on_body(const struct aws_byte_cursor *data, bool finished,
     AWS_LOGF_TRACE(
         AWS_LS_HTTP_STREAM, "id=%p: Incoming body: %zu bytes received.", (void *)&incoming_stream->base, data->len);
 
-    connection->thread_data.body_bytes_decoded += data->len;
+    if (connection->base.manual_window_management) {
+        /* Let stream window shrink by amount of body data received */
+        AWS_ASSERT(incoming_stream->thread_data.window_size >= data->len); /* TODO: assert or real check?*/
```

Comment:
I vote for assert here. The length of data should never be larger than the stream window size here, and we ensure that by only pass data less than stream window size into decoder

Reply:
I went with: assert, but also saturate in case I'm wrong